### ISSUE TITLE
Registered custom editor not used for nested property path with double bracket syntax

### ISSUE THREAD

**Issue by gseifert** [thymeleaf/thymeleaf-spring#109]
Using Thymeleaf 2.1.4, if I register a custom editor for a nested property path such as:
`binder.registerCustomEditor(Date.class, "example.modDt", new CustomDateEditor(dateFormat, true));`
and in my thymeleaf template I use the double bracket syntax with a th:text attribute for example:
`<p data-th-text="*{{example.modDt}}">20160511 12:34:56.789</p>`
it prints using my configured date format in the conversion service instead of the custom format for this field.  If I use th:field such as:
`<input type="hidden" data-th-field="*{example.modDt}" />`
then it prints using the registered custom editor.
Looks like the issue is in the `FieldUtils.isBound()` and `FieldUtils.validateBeanPath()` methods where it only takes into account 1 dot, the one after the form object, but not any dots in nested property paths.  Taking a quick glance at the FieldUtils class in the Thymeleaf 3 code, it looks like the same code.

**Comment by danielfernandez** [comment]
Confirmed and fixed. 

Actually, after evaluating this carefully I noticed something interesting: the fact that we were checking property paths incorrectly here made Thymeleaf return the same results as Spring's own `<s:eval>` JSP tag, which makes use of the `ConversionService` but does not ask at all for any form-bound-object `PropertyEditor`s. 

So my first reaction to this was _"nah, this is fine"_ until I verified that we should indeed be offering additional support for these `PropertyEditor`s when the variable involved in the double-brace expression is a form-bound object.
